code/game/objects/effects/portals.dm

```diff
@@ -88,7 +89,7 @@
 	if(!M.simulated || iseffect(M))
 		. = FALSE
 
-	if(M.anchored && ismecha(M))
+	if(!can_mecha_pass && M.anchored && ismecha(M))
```

Comment:
I'm not 100% sure myself why this check is even here. Since in `mecha/Bump` this is written:
![image](https://user-images.githubusercontent.com/15887760/89132570-727c4e00-d515-11ea-8b0b-706de72ea638.png)
This code only happens if you walk into a portal. Which is a bit... dumb that it even exists.

Can you check why this check here exists and if it's still required or not. If not then you can remove both this check and the ugly "fix" in `mecha/Bumb()`